pkg/auth/auth.go

```diff
@@ -132,9 +152,9 @@ func Login(ctx context.Context, systemContext *types.SystemContext, opts *LoginO
 		return errors.Wrap(err, "getting username and password")
 	}
 
-	if err = docker.CheckAuth(ctx, systemContext, username, password, server); err == nil {
+	if err = docker.CheckAuth(ctx, systemContext, username, password, key); err == nil {
```

Comment:
AFAICS `CheckAuth` does not work with non-registry arguments, compare https://github.com/containers/common/pull/659#discussion_r662283260 .

Reply:
Reverted that change.